utils/pstor/src/products/mod.rs

```diff
@@ -0,0 +1,29 @@
+use crate::{api::Store, common::StorableObjectType};
+use tracing::info;
+
+pub mod v1;
+pub mod v2;
+
+/// Migrate the values from legacy keys to latest keys.
+pub async fn migrate_product_v1_to_v2<S: Store>(
+    store: &mut S,
+    spec_type: StorableObjectType,
+) -> Result<(), crate::Error> {
+    info!("Migrating {spec_type:?} from legacy to current key space");
+    let prefix = &v1::key_prefix_obj(spec_type);
+    let store_entries = store.get_values_prefix(prefix).await?;
+    for (k, v) in &store_entries {
+        // Bail if we are not able extract the resource id. This can leave
+        // incomplete migration entries in etcd, if failed midway.
+        let id = k
+            .split('/')
+            .last()
+            .ok_or_else(|| crate::Error::InvalidKey { key: k.to_string() })?;
+
+        store
+            .put_kv(&format!("{}/{}", v2::key_prefix_obj(spec_type, 0), id), v)
```

Comment:
api version (0) should be passed as arg to the fn ?

Reply:
Done.